xarray/core/variable.py

```diff
@@ -42,7 +42,7 @@ def as_variable(obj, name=None, copy=False):
         obj = obj.copy(deep=False)
     elif hasattr(obj, 'dims') and (hasattr(obj, 'data') or
                                    hasattr(obj, 'values')):
-        obj = Variable(obj.dims, getattr(obj, 'data', obj.values),
+        obj = Variable(obj.dims, getattr(obj, 'data', getattr(obj, 'values')),
```

Comment:
Yeah I think `getattr` is only useful if you don't know what `attr` it might be. But no harm!